Make http.Response.Status respect RFC 2616, like in Go stdlib.

constantoine opened the pull request:
Although RFC 2616 has been superseded with RFC 9110, some client implementation still rely on certain behavious regarding the Status-Line.

Now, as httpmock currently implements it, per [RFC 9110#6.2-1](https://www.rfc-editor.org/rfc/rfc9110.html#section-6.2-1), the "reason phrase" is actually optional, and you need only to include the status code.

Back in the days of [RFC 2616#6.1](https://www.rfc-editor.org/rfc/rfc2616.html#section-6.1), the Status-Line had to contain both the status code (eg. 200) and its textual representation (eg. OK).

Change code is directly stolen from the Go standard library.

This aims to fix #83 .

maxatome commented on response.go:
```diff
@@ -525,7 +525,7 @@ func NewNotFoundResponder(fn func(...any)) Responder {
 func NewStringResponse(status int, body string) *http.Response {
 	return &http.Response{
-		Status:        strconv.Itoa(status),
+		Status:        fmt.Sprintf("%d %s", status, http.StatusText(status)),
```
RFC2616 says that the status-code is a 3-digit integer. As the status comes from outside, perhaps should we use:
```suggestion
		Status:        fmt.Sprintf("%03d %s", status, http.StatusText(status)),
```
instead?
It doesn't protect us from negative and >999 statuses, but at least <100 ones conform to the RFC.
Note that http/httptest uses this format.

constantoine replied:
This absolutely works for me. As far as implementation goes, I looked up Go's http, but http/httptest is just as valuable a source.

maxatome commented:
Thanks @constantoine, could you squash your commits please?

constantoine committed:
Make http.Response.Status respect RFC 2616, like in Go stdlib.

Although RFC 2616 has been superseded with RFC 9110, some client implementation still rely on certain behavious regarding the Status-Line.

Now, as httpmock currently implements it, per [RFC 9110#6.2-1](https://www.rfc-editor.org/rfc/rfc9110.html#section-6.2-1), the "reason phrase" is actually optional, and you need only to include the status code.

Back in the days of [RFC 2616](https://www.rfc-editor.org/rfc/rfc2616.html#section-6.1), per #6.1, the Status-Line had to contain both the status code (eg. 200) and its textual representation (eg. OK).

Change code is directly stolen from the Go standard library.

Signed-off-by: Cléo Rebert <cleo.rebert@gmail.com>

constantoine commented:
@maxatome I think we should be good :)

maxatome commented:
Thanks @constantoine! I hope to publish a new release soon, I work on a new feature not fully ready yet.
